tsdb/engine/tsm1/engine.go

```diff
@@ -203,6 +204,69 @@ func (e *DevEngine) LoadMetadataIndex(_ *tsdb.Shard, index *tsdb.DatabaseIndex,
 	return nil
 }
 
+// Backup will write a tar archive of any TSM files modified since the passed
+// in time to the passed in writer. The basePath will be prepended to the names
+// of the files in the archive. It will force a snapshot of the WAL first
+// then perform the backup with a read lock against the file store. This means
+// that new TSM files will not be able to be created in this shard while the
+// backup is running. For shards that are still acively getting writes, this
+// could cause the WAL to backup, increasing memory usage and evenutally rejecting writes.
+func (e *DevEngine) Backup(w io.Writer, basePath string, since time.Time) error {
```

Comment:
Ack, makes sense to run this on demand as opposed to running during primetime hours.